core/src/main/java/cz/xtf/core/openshift/CLIUtils.java

```diff
@@ -43,7 +44,12 @@ public static String executeCommand(Map<String, String> environmentVariables, St
                 }
             });
 
-            int result = p.waitFor();
+            // Wait for command 10 seconds
+            p.waitFor(10, TimeUnit.SECONDS);
```

Comment:
it could be used for long running actions. I see usages like
```
CLIUtils.executeCommand(podmanCommand(), "build", "-t", IMAGE_NAME, "-f", ...
```
podman build can easily take longer than 10 seconds. There must be some better way to align termination of Process and ExecutorService.

Reply:
Hi @simkam The `waitFor` is not a root cause. So we can keep it without any timeout. But we have to close threads for Output and Error stream. This will solve my issue.